### PR TITLE
850: Bulk Award Assignment

### LINKED ISSUE #850: Assign Award to Multiple Personnel at one time
Just some basic quality of life here. If you deploy an infantry platoon its a pain to award everyone their ribbon for that deployment.

### PR DESCRIPTION
**Pull request by Windchild292**
This fixes #850.

### CODE REVIEW

File: MekHQ/src/mekhq/campaign/personnel/PersonAwardController.java [hunk 1]
```diff
@@ -138,27 +139,48 @@ public void addAwardFromXml(Award award){
...
 
+    public void removeAward(String setName, String awardName, String stringDate) {
+        if (StringUtil.isNullOrEmpty(stringDate)) {
+            MekHQ.getLogger().error(getClass(), "removeAward",
+                    "Attempted to remove award with a null date. Returning without removing the award");
+        } else {
+            removeAward(setName, awardName, stringDate, null);
+        }
+    }
     /**
```
Comment (sixlettervariables) [inline MekHQ/src/mekhq/campaign/personnel/PersonAwardController.java:150]:
```suggestion

    /**
```

File: MekHQ/src/mekhq/gui/adapter/PersonnelTableMouseAdapter.java [hunk 2]
```diff
@@ -1836,95 +1845,97 @@ private void maybeShowPopup(MouseEvent e) {
...
-                        menuItem.setEnabled(person.getXp() >= cost);
-                        if(person.hasSkill(type)) {
-                            currentMenu.add(menuItem);
+                } else {
+                    Set<Award> awards = new TreeSet<>((a1, a2) -> {
+                        if (a1.getSet().equalsIgnoreCase(a2.getSet()) && a1.getName().equalsIgnoreCase(a2.getName())) {
+                            return 0;
                         } else {
-                            newMenu.add(menuItem);
+                            return 1; // This is purposeful, we want to keep the original copy
```
Comment (sixlettervariables) [inline MekHQ/src/mekhq/gui/adapter/PersonnelTableMouseAdapter.java:1918]: This is eventually going to break something as it doesn't follow the invariants. Why not sort by set then name?

I also don't follow the comment about the "original copy".

Comment (Windchild292) [reply MekHQ/src/mekhq/gui/adapter/PersonnelTableMouseAdapter.java:1918]: What, how... That was fixed 🤔 

Oh, I forgot to push the last commit 😳

Comment (Windchild292) [reply MekHQ/src/mekhq/gui/adapter/PersonnelTableMouseAdapter.java]: Pushed, sorry about that. (it did break something, funnily enough)

**Comment by Windchild292** [comment]
This required some changes to fix the unit testing, which has now been resolved.

**Pull request merged**